whisper_experiments/utils/string_utils.py

```diff
@@ -0,0 +1,329 @@
+
+        Notes
+        -----
+        If wrapped text is modified, it cannot be decided which content to return,
+        i.e. the left or the right version.
+        """
+        return getattr(self.text_diff, "content", None)
+
+    @property
+    def content_before(self) -> Optional[str]:
+        """
+        Returns
+        -------
+        Optional[str]
+            The left version of the wrapped text.
+            None if is_added() == True
+        """
+        if self.is_removed:
+            return self.content
+        return getattr(self.text_diff, "content_before", None)
+
+    @property
+    def content_after(self) -> Optional[str]:
+        """
+        Returns
+        -------
+        Optional[str]
+            The right version of the wrapped text.
+            None if is_removed() == True
+        """
+        if self.is_added:
+            return self.content
+        return getattr(self.text_diff, "content_after", None)
+
+
+class LineComparison(NamedTuple):
+    # Union[ModifiedLine, RemovedLine, AddedLine] wrapped in TextDiff
+    line: TextDiff
+    # List of different words in this line
+    words: List[TextDiff]
+
+    def __str__(self) -> str:
+        words_str = ", ".join(map(str, self.words))
+        return (
+            f"    line: {self.line}\n"
+            f"    words: [{words_str}]\n"
+        )
+
+
+class TextComparison(NamedTuple):
+    # Similarity score between the left and the right text blobs
+    similarity: float
+    # List of different lines
+    lines: List[LineComparison]
+
+    def __eq__(self, other):
+        return self.similarity == other.similarity and self.lines == other.lines
+
+    def __str__(self) -> str:
+        lines_str = "\n".join(map(str, self.lines))
+        return (
+            f"similarity: {self.similarity}\n"
+            f"lines: [\n{lines_str}\n]"
+        )
+
+
+def _is_unchanged(text_diff: TextType) -> bool:
+    """
+    Helper function to filter out text does not change from left to right.
+    """
+    return isinstance(text_diff, UnchangedLine)
+
+
+def word_differences(
+    words_1: Iterable[str], words_2: Iterable[str]
+) -> Iterator[TextDiff]:
+    """
+    Return list of removed/added/modified words bewteen the given lists of words.
+
+    Parameters
+    ----------
+    words_1: Iterable[str]
+        Left list of words
+    words_2: Iterable[str]
+        Right list of words
+
+    Yields
+    ------
+    TextDiff
+        Union[RemovedWord, AddedWord, ModifiedWord] wrapped as TextDiff
+
+    See Also
+    --------
+    text_diff.text_differences()
+
+    Notes
+    -----
+    Unchanged words are excluded.
+    """
+    diff_words = text_diff.text_differences(words_1, words_2).diff_lines
+    diff_words = filterfalse(_is_unchanged, diff_words)
+    return map(TextDiff, diff_words)
+
+
+def line_differences(
+    lines_1: Iterable[str],
+    lines_2: Iterable[str],
+    word_split_func: Optional[WordSplitter] = None,
```

Comment:
I think the typing here could be `word_split_func: Callable = str.split`

then you can just do `word_split_func()` because it has already been set to the default.